pandas/tests/io/test_sql.py

```diff
@@ -1357,10 +1357,17 @@ def test_not_reflect_all_tables(self):
 
     def test_warning_case_insensitive_table_name(self, test_frame1):
         # see gh-7815
-        #
-        # We can't test that this warning is triggered, a the database
-        # configuration would have to be altered. But here we test that
-        # the warning is certainly NOT triggered in a normal case.
```

Comment:
@jorisvandenbossche since there is `check_case_insensitive`, I think it is possible to test the warning without altering the database?

This text was added in https://github.com/pandas-dev/pandas/pull/8180/files

Reply:
That's from a long time ago ;)

Yes, we can certainly test the helper method to raise the warning (as you did now), that's useful to add, but strictly speaking that's still not a 100% guarantee that it will work in practice, as it doesn't ensure we call this helper method (we could accidentally remove that in a refactor). 
(maybe we could also mock the helper to ensure it's being called from to_sql .. but I think the current test is also good enough)